newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -528,6 +541,51 @@ export default class AuthenticatedUserProvider extends React.Component<
     }
   };
 
+  _notifyUserAboutEmailVerificationAndAdditionalInfo = () => {
+    const { profile } = this.state.authenticatedUser;
+    if (!profile) return;
+    // If the user has not verified their email when logging in we show a dialog to do so.
+    // - If they just registered, we don't send the email again as it will be sent automatically,
+    // nor do we show a button to send again.
+    // - If they are just logging in, we don't send the email but we show a button to send again.
+    // Use a state value to show the dialog only once.
+    const accountAgeInMs = Date.now() - profile.createdAt;
+    const hasJustCreatedAccount = accountAgeInMs < 1000 * 10; // 10 seconds.
+    if (
+      this.state.authenticatedUser.firebaseUser &&
+      !this.state.authenticatedUser.firebaseUser.emailVerified &&
+      !this._hasNotifiedUserAboutEmailVerification
+    ) {
+      setTimeout(
+        () =>
+          this.openEmailVerificationDialog({
+            open: true,
+            sendEmailAutomatically: false,
+            showSendEmailButton: !hasJustCreatedAccount,
+          }),
+        1000
+      );
+    } else {
+      // If the user has not filled additional info, we show a dialog to do so.
+      this._notifyUserAboutAdditionalInfo();
+    }
+  };
+
+  _notifyUserAboutAdditionalInfo = () => {
+    const profile = this.state.authenticatedUser.profile;
+    if (!profile) return;
+    // If the user has not filled their additional information, show
+    // the dialog to fill it, but ensure they have closed the email verification dialog first.
+    // Use a state value to show the dialog only once.
```

Comment:
That's not really a state value, it it?

Reply:
kinda is, using this.xxx use the React state internally I believe. I'll update the wording if that's confusing